README.md

```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
 
 ## Usage
 
-The [``amazon.cloud collection``](https://github.com/ansible-collections/amazon.cloud) modules are built using:
-
-```tox -e refresh_modules```
-
-The modules will be generated in the `cloud` subdirectory by default. A different target directory can be specified by using:
-
-```tox -e refresh_modules --target-dir /somewhere/else```
+The `amazon_cloud_code_generator`'s core content is hosted in the `amazon_cloud_code_generator/` directory containing the following subdirectories:
+
+- `cmd/` - hosts the main logic of the `amazon_cloud_code_generator`.
+    ```
+    │   ├── generator.py  # Implements the logic to scrape the CloudFormation resource schema and performs all the pre-processing needed to produce each module's DOCUMENTATION block
+    │   ├── refresh_examples.py  # Implements the logic to update EXAMPLES block scraping all the tasks tagged with `tags: [docs]` from `amazon_cloud_code_generator/data/tests/integration/targets/`
+    │   ├── refresh_modules.py  # Implements the logic to generate all the modules specified inside `resources.py` including all the utilities
+    │   ├── resources.py  # Sets the AWS resource names we are going to generate  
+    │   ├── utils.py  # Implements utility functions for the generator
+    ```
+
+- `config/` - Contains the `modules.yaml` file where, for each of the AWS resources we generate, an entry representing the module's name must be added. Additionally, you can also set other information that will be used in module's DOCUMENTATION like short_description, description, and so forth.
+
+- `data/` - Contains the content that will be copied over the generated collection. A high level overview of the subdirectory's content is shown below.
+    ```
+    ├── changelogs  # Changelog fragments that will be used for the new collection release
+    │   ├── plugins
+    │   │   └── module_utils # Modules utils that are used by the collection plugins
+    │   │       ├── core.py
+    │   │       ├── utils.py
+    │   │       └── waiters.py
+    │   └── tests  # Contains the units and integrations tests
+    │       ├── config.yml
+    │       ├── integration
+    │       │   ├── constraints.txt
+    │       │   ├── requirements.txt
+    │       │   └── targets
+    │       │       ├── logs
+    │       │       │   ├── aliases
+    │       │       │   └── tasks
+    │       │       │       └── main.yml
+    │       │       ├── ...
+    │       └── unit
+    │           ├── constraints.txt
+    │           ├── module_utils
+    │           │   ├── test_core.py
+    │           │   └── test_utils.py
+    │           └── requirements.txt
+    ```
+
+- `templates/` - Hosts the templates files that will be used to generate each resource's plugin. 
+    ```
+    ├── default_module.j2
+    └── header.j2
+    ```
+
+The modules generated by the `amazon_cloud_code_generator/` will be made available in the `cloud` subdirectory by default. You can use the following command to do so:
```

Comment:
```suggestion
The modules generated by the `amazon_cloud_code_generator/` are available in the `cloud` subdirectory by default. You can use the following command to do so:
```

Reply:
IMO that last sentence is not very clear. Does that provide an example of how to invoke modules generated in the `cloud` subdirectory?

If so, then maybe change "You can use the following command to do so:" to something like "You can invoke modules in that subdirectory as in the following example command:"

Reply:
@oraNod Tried to rephrase. That command allows you to use the amazon_cloud_code_generator basically.